--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,13 @@ cmake_dependent_option(GLFW_BUILD_WIN32 "Build support for Win32" ON "WIN32" OFF)
 cmake_dependent_option(GLFW_BUILD_COCOA "Build support for Cocoa" ON "APPLE" OFF)
 cmake_dependent_option(GLFW_BUILD_X11 "Build support for X11" ON "UNIX;NOT APPLE" OFF)
-cmake_dependent_option(GLFW_BUILD_WAYLAND "Build support for Wayland" ON "UNIX;NOT APPLE" OFF)
+if (UNIX AND NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+    set(GLFW_WAYLAND_DEFAULT ON)
+else()
+    set(GLFW_WAYLAND_DEFAULT OFF)
+endif()
+cmake_dependent_option(GLFW_BUILD_WAYLAND "Build support for Wayland" ${GLFW_WAYLAND_DEFAULT}
+                       "UNIX;NOT APPLE" OFF)
 
 cmake_dependent_option(GLFW_USE_HYBRID_HPG "Force use of high-performance GPU on hybrid systems" OFF
                        "WIN32" OFF)
